src/three-components/SmoothControls.ts

```diff
@@ -154,27 +154,30 @@ export class Damper {
   update(
       x: number, xGoal: number, timeStepMilliseconds: number,
       xNormalization: number): number {
-    if (timeStepMilliseconds >= DECAY_MILLISECONDS) {
-      // This clamps at the point where a large time step would cause the
-      // discrete step to overshoot.
-      this[$velocity] = 0;
+    if (!x) {
```

Comment:
`!x` will be true if `x` is `0`. In most cases, you should use `x == null` for cases where you expect `x` to be unset. If `x` is ever `NaN`, it's usually a bug / code hygiene issue.

Reply:
I see, the error was that x was undefined, which is different than null, right? I'll see if I can track that down. Is there a good way to check that x is just a valid number? Do we have an isFinite() function?

Reply:
`==` and `===` in JavaScript are different, and the nature of `==` means `x == null` will catch both `null` and `undefined`

this-is-fine.jpg